tailnet/coordinator.go

```diff
@@ -146,14 +168,36 @@ type core struct {
 	// nodes maps agent and connection IDs their respective node.
 	nodes map[uuid.UUID]*Node
 	// agentSockets maps agent IDs to their open websocket.
-	agentSockets map[uuid.UUID]*TrackedConn
+	agentSockets map[uuid.UUID]Enqueueable
 	// agentToConnectionSockets maps agent IDs to connection IDs of conns that
 	// are subscribed to updates for that agent.
-	agentToConnectionSockets map[uuid.UUID]map[uuid.UUID]*TrackedConn
+	agentToConnectionSockets map[uuid.UUID]map[uuid.UUID]Enqueueable
+
+	// clients holds a map of all clients connected to the coordinator. This is
+	// necessary because a client may not be subscribed into any agents.
+	clients map[uuid.UUID]Enqueueable
+	// clientsToAgents is an index of clients to all of their subscribed agents.
+	clientsToAgents map[uuid.UUID]map[uuid.UUID]struct{}
```

Comment:
this could be a `map[uuid.UUID]map[uuid.UUID]Enqueueable` and then when you get a client update, you don't have to do a separate lookup in the `agentSockets` to get the socket.